tests/test_builtin_casters.cpp

```diff
@@ -378,4 +378,21 @@ TEST_SUBMODULE(builtin_casters, m) {
     m.def("takes_const_ref", [](const ConstRefCasted &x) { return x.tag; });
     m.def("takes_const_ref_wrap",
           [](std::reference_wrapper<const ConstRefCasted> x) { return x.get().tag; });
+
+    // test return_value_policy::return_as_bytes
+    m.def(
+        "invalid_utf8_string_as_bytes",
+        []() { return std::string("\xba\xd0\xba\xd0"); },
```

Comment:
```suggestion
        []() { return py::bytes(std::string("\xba\xd0\xba\xd0")); },
```
is all that was ever needed.